@escherize Looks like this breaks launching the Uberjar at initialization.

To reproduce: `./bin/build` and then run it `java -jar target/uberjar/metabase.jar`:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:416)
        at clojure.lang.RT.classForName(RT.java:2212)
        at clojure.lang.RT.classForName(RT.java:2221)
        at clojure.lang.RT.loadClassForName(RT.java:2240)
        at clojure.lang.RT.load(RT.java:449)
        at clojure.lang.RT.load(RT.java:424)
        at clojure.core$load$fn__6856.invoke(core.clj:6115)
        at clojure.core$load.invokeStatic(core.clj:6114)
        at clojure.core$load.doInvoke(core.clj:6098)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.lang.Var.invoke(Var.java:384)
        at clojure.lang.Util.loadWithClass(Util.java:251)
        at metabase.core.<clinit>(Unknown Source)
Caused by: java.lang.IllegalStateException: Attempting to call unbound fn: #'metabase.util.i18n.impl/locale
        at clojure.lang.Var$Unbound.throwArity(Var.java:45)
        at clojure.lang.AFn.invoke(AFn.java:32)
        at clojure.lang.Var.invoke(Var.java:384)
        at clojure.lang.LispReader$CtorReader.readTagged(LispReader.java:1448)
        at clojure.lang.LispReader$CtorReader.invoke(LispReader.java:1427)
        at clojure.lang.LispReader$DispatchReader.invoke(LispReader.java:846)
        at clojure.lang.LispReader.read(LispReader.java:285)
        at clojure.lang.LispReader.read(LispReader.java:216)
        at clojure.lang.LispReader.read(LispReader.java:205)
        at clojure.lang.RT.readString(RT.java:1879)
        at clojure.lang.RT.readString(RT.java:1874)
        at metabase.core__init.__init0(Unknown Source)
        at metabase.core__init.<clinit>(Unknown Source)```

--- a/src/metabase/util/i18n/impl.clj
+++ b/src/metabase/util/i18n/impl.clj
@@ -190,9 +190,16 @@
                (f)))))))))
 
 (defn site-locale-from-setting
-  "Fetch the value of the `site-locale` Setting."
+  "Fetch the value of the `site-locale` Setting.
+  When metabase is shutting down, we need to log some messages after the db connection is closed, so we keep around a
+  cached-site-locale for that purpose."
   []
-  (@site-locale-from-setting-fn))
+  (let [cached-site-locale (atom "en")]
+    (try
+      (let [site-locale (@site-locale-from-setting-fn)]
+        (reset! cached-site-locale site-locale)
+        site-locale)
+      (catch Exception e @cached-site-locale))))
 
 (defmethod print-method Locale
   [locale ^java.io.Writer writer]

